afids_utils/transforms.py

```diff
@@ -0,0 +1,32 @@
+"""Methods for transforming between different coordinate systems"""
+from __future__ import annotations
+
+from typing import Optional
+
+import numpy as np
+from numpy.typing import NDArray
+
+
+def afid_world2voxel(
+    afid_world: NDArray[np.float_],
+    nii_affine: NDArray[np.float_],
+    resample_size: float = 1,
+    padding: Optional[int] = None,
+) -> NDArray[np.int_]:
+    """Transform fiducials in world coordinates to voxel coordinates
+
+    Optionally, resample to match resampled image
+    """
+
+    # Translation
+    afid_voxel = afid_world.T - nii_affine[:3, 3:4]
+    # Rotation
+    afid_voxel = np.dot(afid_voxel, np.linalg.inv(nii_affine[:3, :3]))
+
+    # Round to nearest voxel
+    afid_voxel = np.rint(np.diag(afid_voxel) * resample_size)
+
+    if padding:
```

Comment:
I kind of think we should just get rid of this arg. It's really straightforward to just pad the output `afid_voxel` if you want and it doesn't really have anything to do with converting to voxel coordinates as far as I can tell.

Reply:
I'm not opposed to this, we can look at reimplementing it in the future if needed. We don't use it anywhere anymore, though I think it use to be part of the original matlab implementation.